src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py

```diff
@@ -755,10 +809,12 @@ def snapshot_info(self, snapname):
                           'data_pool':'ceph.dir.layout.pool'}
             for key, val in snap_attrs.items():
                 snap_info[key] = self.fs.getxattr(snappath, val)
-            return {'size': int(snap_info['size']),
+            pending_clone_dict = self.get_pending_clones(snapname)
```

Comment:
I think keeping `has_pending_clones` independent of the `pending_clone_dict` is better. As in, you need to still show `has_pending_clones` if there are no clones created.

```
if self.has_pending_clones(snapname):
   pending_clones = "yes"
   pending_clone_dict = self.get_pending_clones(snapname)
else:
   pending_clones = "no"
...
...
'has_pending_clones': pending_clones
```

Reply:
I have kept `has_pending_clones` and `pending_clones` in this function because they are closely related.
Finding list of `pending_clones` is depending on result of `has_pending_clones`.

Reply:
I have kept `has_pending_clones` and `pending_clones` in this function because they are closely related.
Finding list of `pending_clones` is depending on result of `has_pending_clones`.
I am showing `has_pending_clones` if there are no clones created. I have also added test for this.